packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

```diff
@@ -15,6 +15,20 @@ import { getLayoutType } from '../../layouts';
 
 const pendingSettingsUpdates = new WeakMap();
 
+function useShallowMemo( value ) {
+	const ref = useRef();
+
+	// Store current value in ref if it is not shallow equal to the previous value.
+	useEffect( () => {
+		if( ! isShallowEqual( ref.current, value ) ) {
+			ref.current = value
```

Comment:
The problem is that, this won't "re-render" the component, so the component is going to use the old value unless there's something else in the component that causes a "re-render".

Now, we could replace this with a state value instead of a ref value but it also has a small downside: it will render the component twice: once with the old value and once with the new value. I wonder if there's a way to avoid the two renders.

Reply:
Maybe we could use a state update during the render; that should avoid rendering twice. Cc @kevin940726

Ref: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

P.S. The `isShallowEqual` import is missing from the file.

## Example needs testing
```js
function useShallowMemo( value ) {
	const [ prevValue, setPrevValue ] = useState( value );

	if ( ! isShallowEqual( prevValue, value ) ) {
		setPrevValue( value );
	}

	return prevValue;
}
```